memflow-win32/examples/keyboard_listen.rs

```diff
@@ -0,0 +1,110 @@
+/*!
+This example shows how to use a dynamically loaded connector in conjunction
+with memflow-win32 to read the key-states of a windows computer.
+This example uses the `Inventory` feature of memflow but hard-wires the
+connector instance into the memflow-win32 OS layer.
+
+The example is an adaption of the memflow core process list example:
+https://github.com/memflow/memflow/blob/next/memflow/examples/process_list.rs
+
+# Usage:
+```bash
+cargo run --release --example keyboard_listen -- -vv -c kvm
+```
+*/
+use clap::*;
+use log::Level;
+
+use memflow::prelude::v1::*;
+use memflow_win32::prelude::v1::*;
+
+pub fn main() -> Result<()> {
+    let matches = parse_args();
+    let (chain, conn_args) = extract_args(&matches)?;
+
+    // create inventory + connector
+    let inventory = Inventory::scan();
+    let connector = inventory
+        .builder()
+        .connector_chain(chain)
+        .args(conn_args.parse()?)
+        .build()?;
+
+    let os = Win32Kernel::builder(connector)
+        .build_default_caches()
+        .build()
+        .unwrap();
+
+    let mut kb = os.into_keyboard()?;
+    println!("Running keyboard listener...\n...............................");
+    println!("Press ESC to exit");
+    println!("Press LSHIFT to see if it is down");
+    // listen for keyboard events until escape is pressed
+    while !kb.is_down(vkey::VK_ESCAPE.into()) {
+        if kb.is_down(vkey::VK_LSHIFT.into()) {
+            println!("Left Shift is down");
+        }
+    }
+
+    Ok(())
+}
+
+fn parse_args() -> ArgMatches {
+    Command::new("process_list example")
+        .version(crate_version!())
+        .author(crate_authors!())
+        .arg(Arg::new("verbose").short('v').action(ArgAction::Count))
+        .arg(
+            Arg::new("connector")
+                .short('c')
+                .action(ArgAction::Append)
+                .required(true),
+        )
+        .arg(
+            Arg::new("connector_args")
+                .short('a')
+                .action(ArgAction::Append)
+                .required(false),
+        )
+        .arg(Arg::new("os").short('o').action(ArgAction::Append))
+        .get_matches()
+}
+
+fn extract_args(matches: &ArgMatches) -> Result<(ConnectorChain<'_>, String)> {
+    let log_level = match matches.get_count("verbose") {
+        0 => Level::Error,
+        1 => Level::Warn,
+        2 => Level::Info,
+        3 => Level::Debug,
+        4 => Level::Trace,
+        _ => Level::Trace,
+    };
+    simplelog::TermLogger::init(
```

Comment:
I think we should switch to env_logger like most other memflow crates now use. But this is probably outside of the scope of this pr.

Reply:
ok we can open a new issue for that then